.github/workflows/plugin-check.yml

```diff
@@ -0,0 +1,39 @@
+name: WordPress Plugin Checks
+
+on:
+  push:
+    branches:
+      - develop
+      - trunk
+  pull_request:
+    branches:
+      - develop
+
```

Comment:
We'll want to ensure we're setting proper permissions here. The approach we've taken so far is we disable all permissions at the workflow level by doing `permissions: {}` and then we set the proper permissions for each job we run (in this case the `build` and `test` jobs will need those set)